mlflow/gateway/providers/openai.py

```diff
@@ -121,18 +133,214 @@ async def chat_stream(
+                message="Databricks SDK is required to use UC functions",
+                error_code=404,
+            )
+
+        warehouse_id = os.environ.get("DATABRICKS_WAREHOUSE_ID")
+        if warehouse_id is None:
+            raise HTTPException(
+                status_code=400,
+                detail="DATABRICKS_WAREHOUSE_ID environment variable is not set",
+            )
+
+        from fastapi.encoders import jsonable_encoder
+
+        payload = jsonable_encoder(payload, exclude_none=True)
+        self.check_for_model_field(payload)
+
+        workspace_client = WorkspaceClient()
+
+        user_msg = payload["messages"][0]["content"]
+
+        token_usage_accumulator = TokenUsageAccumulator()
+        tool_calls, tool_messages, parsed = parse_uc_functions(payload["messages"][0]["content"])
+        if tool_calls:
+            user_tool_messages = [m for m in payload["messages"] if m["role"] == "tool"]
+            user_tool_calls = next(m for m in payload["messages"] if "tool_calls" in m).get(
+                "tool_calls"
+            )
+            messages = [
+                {
+                    "role": "user",
+                    "content": parsed,
+                },
+                {
+                    "role": "assistant",
+                    "content": None,
+                    "tool_calls": tool_calls + user_tool_calls,
+                },
+                *tool_messages,
+                *user_tool_messages,
+            ]
+            resp = await send_request(
+                headers=self._request_headers,
+                base_url=self._request_base_url,
+                path="chat/completions",
+                payload=self._add_model_to_payload_if_necessary(
+                    {
+                        **payload,
+                        "messages": messages,
+                    }
+                ),
+            )
+
+        elif any(t["type"] == "uc_function" for t in payload.get("tools", [])):
+            updated_tools = []
+            uc_func_mapping = {}
+            for tool in payload.get("tools", []):
+                if tool["type"] == "uc_function":
+                    function_name = tool["uc_function"]["name"]
+                    function = workspace_client.functions.get(function_name)
+                    param_metadata = get_func_schema(function)
+                    t = {
+                        "type": "function",
+                        "function": param_metadata,
+                    }
+                    uc_func_mapping[t["function"]["name"]] = function
+                    updated_tools.append(t)
+                    continue
+                else:
+                    updated_tools.append(tool)
+
+            payload["tools"] = updated_tools
+
+            messages = payload.pop("messages")
+            uc_func_calls = []
+            user_tool_calls = []
+            resp = None
+            should_break = False
+            for _ in range(20):
```

Comment:
Just checking - is this consistent with the behavior on Databricks (max 20 tool calls)?